src/lib/jsonParseV2/parseEntityVector.cpp

```diff
@@ -53,11 +53,16 @@ std::string parseEntityVector(ConnectionInfo* ciP, const Value::ConstMemberItera
 
   for (Value::ConstValueIterator iter2 = iter->value.Begin(); iter2 != iter->value.End(); ++iter2)
   {
+
     std::string  r;
     Entity*      eP = new Entity();
 
     evP->vec.push_back(eP);
 
+    // FIXME PR: is the logic of this function correct? I mean, parseEntityObject() could result in error, so the push_back maybe
+    // should be done after the parseEntityObject() call (and delete the eP just before return r). Or maybe it doesn't matter... as in the
+    // case of error the error is propagated back to result in an OrionError and the evP is not actually used (and properly destroyed during
+    // the release step)
     r = parseEntityObject(ciP, iter2, eP, attributesAllowed);
```

Comment:
I imagine the push is done before so that evP->release() is able to delete the allocated Entity.
Apart from that, yes, it would be better not to push (and to delete 'explicitly') in case of error.

Can't say I care too much ...